src/ts/utils/clientRect.ts

```diff
@@ -0,0 +1,26 @@
+const topBuffer = 2;
+const leftBuffer = 2;
+
+export const getClientRectIndex = (event: Event): number | undefined => {
```

Comment:
I'd suggest adding a comment for this function to answer the question – get the client rect for what?